x-pack/plugins/apm/server/routes/correlations/queries/query_significant_correlations.ts

```diff
@@ -76,12 +77,51 @@ export const fetchSignificantCorrelations = async (
     )
   );
 
-  const latencyCorrelations: LatencyCorrelation[] = fulfilled.filter(
-    (d): d is LatencyCorrelation => d !== undefined
-  );
+  let fallbackResult: LatencyCorrelation | undefined;
+  const latencyCorrelations: LatencyCorrelation[] = [];
+
+  fulfilled.forEach((d: LatencyCorrelation | undefined) => {
+    if (d === undefined) return;
+    if (Array.isArray(d.histogram)) {
+      latencyCorrelations.push(d);
+    } else {
+      if (!fallbackResult) {
+        fallbackResult = d;
+      } else {
+        if (
+          d.ksTest > fallbackResult.ksTest &&
+          d.correlation > fallbackResult.correlation
+        ) {
+          fallbackResult = d;
+        }
+      }
+    }
+  });
+
+  if (latencyCorrelations.length === 0 && fallbackResult) {
+    const { fieldName, fieldValue } = fallbackResult;
+    const logHistogram = await fetchTransactionDurationRanges(
+      esClient,
+      paramsWithIndex,
+      histogramRangeSteps,
+      [{ fieldName, fieldValue }]
+    );
+
+    if (typeof fallbackResult === 'object' && fallbackResult !== null) {
+      fallbackResult = {
+        ...(fallbackResult as LatencyCorrelation),
```

Comment:
This shouldn't be necessary. The `if` clause should act as a typeguard and narrow the type from option to `LatencyCorrelation`

```suggestion
        ...fallbackResult,
```

Reply:
Updated here https://github.com/elastic/kibana/pull/126683/commits/9f57ca31de5d775db046dfc0b09fcf504fdba73f